src/main/java/org/example/jvspringbootfirstbook/ropsitory/BookRepositoryImpl.java

```diff
@@ -1,19 +1,59 @@
 package org.example.jvspringbootfirstbook.ropsitory;
 
-import org.example.jvspringbootfirstbook.model.Book;
-
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Root;
 import java.util.List;
-import java.util.Optional;
+import org.example.jvspringbootfirstbook.exception.DataProcessingException;
+import org.example.jvspringbootfirstbook.model.Book;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.criteria.HibernateCriteriaBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
 
+@Repository
 public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
 
     @Override
     public Book save(Book book) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(book);
+            transaction.commit();
+        } catch (HibernateException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Unable to save book", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return book;
     }
 
     @Override
     public List<Book> findAll() {
-        return List.of();
+        try (Session session = sessionFactory.openSession()) {
+            HibernateCriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Book> query = criteriaBuilder.createQuery(Book.class);
+            Root<Book> root = query.from(Book.class);
```

Comment:
root variable is not used. Check if it's needed, maybe something is missing?
by adding query.select(root) you can ensure all fields are selected.
Root<Book> root = query.from(Book.class);
            query.select(root);
            return session.createQuery(query).getResultList();